src/components/schedule-visualizer/robot-trajectory.tsx

```diff
@@ -22,36 +28,82 @@ export const RobotTrajectory = React.memo(
   ): React.ReactElement {
     debug('render');
 
-    const { trajectory, conflicts, footprint, ...otherProps } = props;
+    const { trajectory, conflicts, footprint, colorManager, ...otherProps } = props;
     const theme = useTheme();
+    const isConflict = conflicts.flat().includes(trajectory.id);
+    const settings = React.useContext(SettingsContext);
+    const trajDiameter = settings.trajectoryDiameter;
+    const isOutline = settings.trajectoryAnimation === TrajectoryAnimation.Outline;
 
-    const color = React.useMemo(
-      () =>
-        conflicts.flat().includes(trajectory.id)
-          ? theme.palette.error.main
-          : theme.palette.success.main,
-      [trajectory, conflicts, theme],
-    );
+    function determineTrajDiameter(trajDiameter: TrajectoryDiameter): number {
+      switch (trajDiameter) {
+        case TrajectoryDiameter.FixSize:
+          return fixTrajectoryDiameter;
+        case TrajectoryDiameter.RobotSize:
+          return footprint;
+      }
+    }
+
+    const trajectoryDiameter = determineTrajDiameter(trajDiameter);
+
+    const color = React.useMemo(() => {
+      const getRobotColor = () => {
+        const robotColor = colorManager?.robotColorFromCache(
+          trajectory.fleet_name,
+          trajectory.robot_name,
+        );
+        return !!robotColor ? robotColor : theme.palette.success.main;
+      };
+      const getPathColor = () => {
+        const pathColor = colorManager?.pathColorFromCache(
+          trajectory.fleet_name,
+          trajectory.robot_name,
+        );
+        return !!pathColor ? pathColor : theme.palette.success.main;
+      };
+      const robotColorHolder = getRobotColor();
+      const pathColorHolder = getPathColor();
+      switch (settings.trajectoryColor) {
+        case TrajectoryColor.Theme:
+          return theme.palette.success.main;
+        case TrajectoryColor.RobotColor:
+          return robotColorHolder;
+        case TrajectoryColor.Shades:
+          return pathColorHolder;
+      }
+    }, [trajectory, theme, colorManager, settings.trajectoryColor]);
 
     const pathD = React.useMemo(() => {
       return trajectoryPath(trajectory.segments).d;
     }, [trajectory]);
 
     return (
-      <path
-        data-component="RobotTrajectory"
-        ref={ref}
-        d={pathD}
-        stroke={color}
-        opacity={0.8}
-        strokeWidth={footprint * 0.8}
-        strokeLinecap="round"
-        fill={'none'}
-        pathLength={1}
-        strokeDasharray={2}
-        strokeDashoffset={0}
-        {...otherProps}
-      />
+      <>
+        {color && (
+          <path
+            id="robotTrajectoryPath"
+            data-component="RobotTrajectory"
+            ref={ref}
+            d={pathD}
+            stroke={color}
+            opacity={0.8}
+            strokeWidth={isConflict && !isOutline ? trajectoryDiameter - 0.1 : trajectoryDiameter}
```

Comment:
Changed it for now. Only concern I have is that when we have different robots with different diameters, relating to percentage may cause some outlines to be too thin or too thick resulting in an inconsistent look across the UI. We will see how it goes when the actual footprint data for the robot comes in.

Reply:
Yeah, you're right. Ok, in case we need to change it in the future, what do you think about setting the default to 0.1 (as it was) and have a minimum diameter for the component (let's say it should be always greater than 0.2)

Reply:
Sounds good!